draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -210,6 +210,70 @@ WebRTC applications running in a third-party browser execution context (i.e., a
 context that has a different origin than the top-level execution context), or a
 private browser execution context.
 
+Security Considerations {#security}
+======================
+
+mDNS Message Flooding
+---------------------
+
+The implementation of this proposal requires the mDNS querying capability of the
+browser for registering mDNS names or adding remote ICE host candidates with
+such names. It also requires the mDNS responding capability of either the
+browser or the operating platform of the browser for registering, removing or
+resolving mDNS names. In particular,
+
+- the registration of name requires optional probing queries and mandatory
+announcing responses ({{RFC6762}, Section 8}), and this is performed at the
+beginning of ICE gathering;
+
+- the addition of remote ICE host candidates with mDNS names generates mDNS
+queries for names of each candidate;
+
+- the removal of names could happen when the execution context of the ICE agent
+is destroyed in an implementation, and goodbye responses should be sent to
+invalidate records generated by the ICE agent in the local network
+({{RFC6762}}, Section 10.1).
+
+A malicious Web application could flood the local network with mDNS messages by:
+
+- creating execution contexts that create ICE agents and start gathering of
+  local ICE host candidates;
+
+- destroying these local candidates soon after the name registration is done;
+
+- adding fictitious remote ICE host candidates with mDNS names.
+
+
+{{RFC6762}} defines a per-record multicast rate limiting rule, in which a given
+record on a given interface cannot be sent less than one second since its last
+transmission. This rate limiting rule however does not mitigate the above
+attacks, in which new names, hence new records, are constantly created and sent.
+A global rate limit of mDNS messages that can be created by a Web application,
```

Comment:
By 'global', is the intent here that the limit is browser-wide?

Reply:
The intended limit is per-interface for the execution/browsing context, which could be more restrictive as browser-wide, so that one would not be able to work around the RFC per-record limit by creating/querying tons of new records. This 'global' does read ambiguous, and I will spell out the 'per-interface for the execution context' limit in a new commit, or 'browser-wide' if we would like to be more restrictive. I think a rate limit per-interface for a context is sufficient to mitigate the DoS attacked described above.

Reply:
I have updated the description of the rate limit with also cap MUST prescription. PTAL.